feat(linter): support es2025 env

Sysix opened the pull request:
ES2025 was implemented by globals: https://github.com/sindresorhus/globals/pull/256

and the `Iterator` support included here: https://github.com/sindresorhus/globals/issues/266

Should we support es3, es5 and es2015 too? 
https://github.com/sindresorhus/globals/blob/main/data/es3.mjs
https://github.com/sindresorhus/globals/blob/main/data/es5.mjs
https://github.com/sindresorhus/globals/blob/main/data/es2015.mjs

Sysix commented:
* **#8985** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8985?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8985?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8985).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
> Should we support es3, es5 and es2015 too?

Only support when there is a requirement.

Boshen commented:
### Merge activity

* **Feb 9, 10:24 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(linter): support es2025 env (#8985)

ES2025 was implemented by globals: https://github.com/sindresorhus/globals/pull/256

and the `Iterator` support included here: https://github.com/sindresorhus/globals/issues/266

Should we support es3, es5 and es2015 too?
https://github.com/sindresorhus/globals/blob/main/data/es3.mjs
https://github.com/sindresorhus/globals/blob/main/data/es5.mjs
https://github.com/sindresorhus/globals/blob/main/data/es2015.mjs
